spec/support/shared_system_behavior.rb

```diff
@@ -10,10 +12,10 @@
   before { login(user) }
 end
 
-shared_examples 'a page restricted to super admins' do
-  before { visit url }
-
+shared_examples_for 'a page restricted to super admins' do |expected_content = 'Super Admin'|
   context 'when no user is logged in' do
+    before { visit url }
```

Comment:
I've moved the `before { visit url }` call into the relevant specs to avoid an order-of-operations bug. The `user is logged in` context was occasionally being set *after* the page visitation. This adds some slight duplication of `visit url`, but the explicit ordering fixes the spec bugginess.